custom_components/entsoe/coordinator.py

```diff
@@ -161,49 +166,89 @@ async def fetch_prices(self, start_date, end_date):
                     f"Warning the integration doesn't have any up to date local data this means that entities won't get updated but access remains to restorable entities: {exc}."
                 )
 
+    # ENTSO: the async fetch job itself
     def api_update(self, start_date, end_date, api_key):
         client = EntsoeClient(api_key=api_key)
         return client.query_day_ahead_prices(
             country_code=self.area, start=start_date, end=end_date
         )
+    
+    # ENTSO: Return the data for the given date
+    def get_data(self, date):
+        return {k: v for k, v in self.data.items() if k.date() == date.date()}
 
-    async def get_energy_prices(self, start_date, end_date):
-        # check if we have the data already
-        if (
-            len(self.get_data(start_date)) > MIN_HOURS
-            and len(self.get_data(end_date)) > MIN_HOURS
-        ):
-            self.logger.debug(f"return prices from coordinator cache.")
-            return {
-                k: v
-                for k, v in self.data.items()
-                if k.date() >= start_date.date() and k.date() <= end_date.date()
-            }
-        return self.parse_hourprices(await self.fetch_prices(start_date, end_date))
+    # ENTSO: Return the data for today
+    def get_data_today(self):
+        return self.get_data(self.today)
+
+    # ENTSO: Return the data for tomorrow
+    def get_data_tomorrow(self):
+        return self.get_data(self.today + timedelta(days=1))
+
+    # ENTSO: Return the data for yesterday
+    def get_data_yesterday(self):
+        return self.get_data(self.today - timedelta(days=1))
 
+    # SENSOR: Do we have data available for today TODO: remove
     def today_data_available(self):
         return len(self.get_data_today()) > MIN_HOURS
+    
+    # SENSOR: Get the current price
+    def get_current_hourprice(self) -> int:
+        return self.data[dt.now().replace(minute=0, second=0, microsecond=0)]
+
+    # SENSOR: Get the next hour price
+    def get_next_hourprice(self) -> int:
+        return self.data[
+            dt.now().replace(minute=0, second=0, microsecond=0) + timedelta(hours=1)
+        ]
+
+    # SENSOR: Get timestamped prices of today as attribute for Average Sensor
+    def get_prices_today(self):
+        return self.get_timestamped_prices(self.get_data_today())
 
-    # this method is called by each sensor, each complete hour, and ensures the date and filtered hourprices are in line with the current time
+    # SENSOR: Get timestamped prices of tomorrow as attribute for Average Sensor
+    def get_prices_tomorrow(self):
+        return self.get_timestamped_prices(self.get_data_tomorrow())
+
+    # SENSOR: Get timestamped prices of today & tomorrow or yesterday & today as attribute for Average Sensor
+    # TODO: why is there another logic when data is below 48 hrs and does this ever happen? 
```

Comment:
Because this function returns either the values for yesterday and today or if available the values of today and tomorrow.
Here also the same for the TODO questions do not belong in the codebase.